### DISCUSSION TITLE
Liabilities questions

### DISCUSSION THREAD

**Discussion by OXINARF** [firefly-iii/firefly-iii#5971] [Firefly III]
Hi,

First of all, let me thank you for the work on this software, it's great!

I have been working for some time on importing more than 2 years of data into it - I could just do the basics for past data, but I would like to have everything as organized as possible.
Because that has been such a daunting task, I hadn't updated Firefly in a while - I was still at 5.5.12 and updating also required some work on my host to get PHP 8 working. Anyway, I finally got to it last week and I'm now on 5.6.16 (updating to 5.7.1 will now be easier  🙂).

One thing I have been struggling with are liabilities - or better, how to track long-term and short-term loans with friends and family (in both ways).
I have read all the documentation, several issues and discussions (and even looked at code), but it's still not completely clear to me how things work - unfortunately, I think the changes on 5.6 made it even more confusing in my head.

A couple of questions:
- does the Liability Type has any influence on how things work or is it just a UI thing? Does Mortgage/Loan/Debt show up anywhere other than the Liabilities page?
- what implications does the Liability Direction has? I see there are differences on Amount Due and Balance, but I haven't completely understood how it works (especially how it increases/decreases them); it also seems that it behaves differently for the Start amount of Debt - I didn't really understand why.
- on [Opening Balance documentation](https://docs.firefly-iii.org/firefly-iii/support/transaction_types/#opening-balance), it shows as possible to have an initial positive balance on Liabilities, how do you do that? As far as I can see, it's impossible to put a positive initial amount of debt.
- is there any way that withdrawals/deposits between Liabilities and Asset accounts don't count as income/expenses?

Sorry if that's a lot of questions, really just trying to get my around it.

**Comment by JC5** [comment]
Thanks for the compliments!

1. It's just a UI thing.
2. The direction determins whether you owe money or if you are owed money. If you owe somebody money, it's just a negative amount. Any money you pay back fills that particular "hole". But if somebody owes you money, the money you gave them has to come from somewhere. It's a bit of a technical thing.
3. That's right. The transaction type supports it, but not for liabilities.
4. What did you have in mind instead?

**Reply by OXINARF** [reply]
Thank you for the fast answer.

1. Great. I think I can also now confirm that it is only displayed on the Liabilities page and on the source/destination dropdown when converting a transaction.
2. Well if you owe somebody money, it also had to come from somewhere 🙂 I've done some more experiments today and I've compiled a table on how both directions work (see below).
I understand it now, but I find the owe/owed naming quite strange: it's not really about that, but whether you are tracking a loan with fees/interest associated (or, more generically, whether transactions with assets are expenses/revenue and so have impact on amount due). I can't find a good naming for the option now, but I'll let you know if I think on a better choice of words.
3. I wish you could put a positive number - the "Owed" direction works very well for what I want to do (just loans both ways without expenses/revenue/interest), but if at start date I owe someone then it doesn't work. I guess the only option there would be the "Owe" direction but without doing the Liabiliaty-Expense/Revenue transactions.
4. Well on the "Owed" direction, the transactions between Asset and Liability aren't really income/expenses, they are more like transfers, that's why the Amount Due doesn't change. What I don't like is that it artificially increases the income/expenses in reports.

Writing that last sentence actually gave me the idea that if transfers were allowed between Asset and Liability accounts then we wouldn't need the (current) direction option and basically everything would work (would probably want two new options). Firefly is very complex to try this right now, but I might look at developing this on the very long term.

&nbsp;
How Owe/Owed transactions affect balance/amount due/net worth:

<table>
<thead>
  <tr>
    <th></th>
    <th align="center" valign="middle" colspan="3">Owe</th>
    <th align="center" valign="middle" colspan="3">Owed</th>
  </tr>
</thead>
<tbody>
  <tr>
    <th></th>
    <th align="center" valign="middle">Balance</th>
    <th align="center" valign="middle">Amount due</th>
    <th align="center" valign="middle">Net worth</th>
    <th align="center" valign="middle">Balance</th>
    <th align="center" valign="middle">Amount due</th>
    <th align="center" valign="middle">Net worth</th>
  </tr>
  <tr>
    <td valign="middle">Liability to Expense (withdrawal)</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Goes positive</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Goes positive</td>
    <td align="center" valign="middle">Goes negative</td>
  </tr>
  <tr>
    <td valign="middle">Revenue to Liability (deposit)</td>
    <td align="center" valign="middle">Goes positive</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Goes positive</td>
    <td align="center" valign="middle">Goes positive</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Goes positive</td>
  </tr>
  <tr>
    <td valign="middle">Asset to Liability (withdrawal)</td>
    <td align="center" valign="middle">Goes positive</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Doesn't change</td>
    <td align="center" valign="middle">Goes positive</td>
    <td align="center" valign="middle">Doesn't change</td>
    <td align="center" valign="middle">Doesn't change</td>
  </tr>
  <tr>
    <td valign="middle">Liability to Asset (deposit)</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Goes positive</td>
    <td align="center" valign="middle">Doesn't change</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Doesn't change</td>
    <td align="center" valign="middle">Doesn't change</td>
  </tr>
  <tr>
    <td colspan="7"></td>
  </tr>
  <tr>
    <td valign="middle" rowspan="4">Initial debt</td>
    <th align="center" valign="middle" colspan="3">Liability to Initial Balance transaction</th>
    <th align="center" valign="middle" colspan="3">Liability Credit to Liability transaction</th>
  </tr>
  <tr>
    <td align="center" valign="middle">Negative</td>
    <td align="center" valign="middle">Positive</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Positive</td>
    <td align="center" valign="middle">Doesn't change</td>
    <td align="center" valign="middle">Goes positive</td>
  </tr>
  <tr>
    <td align="center" valign="middle" rowspan="2" colspan="3"></td>
    <th align="center" valign="middle" colspan="3">Liability to Initial Balance transaction</th>
  </tr>
  <tr>
    <td align="center" valign="middle">Negative</td>
    <td align="center" valign="middle">Positive</td>
    <td align="center" valign="middle">Goes negative</td>
  </tr>
</tbody>
</table>

**Comment by JC5** [comment]
Thanks for the table, it's really helpful. 

If you loan money to somebody, and the loan is "just there" from Firefly III's perspective, the math doesn't work out. If I start Firefly III with a loan of 1000 to a friend, your net worth would start at -1000, meaning I am lacking 1000. Once the loan is paid back, I would be back at zero. But that's not true, I would have +1000 (the money I got paid back). So in these instances a hidden account is created that provides the 1000 that you then loan to your friend. When your friend pays you back it doesn't go into the hidden account but straight to your asset account. If you start the loan while using Firefly III, the 1000 is taken from your asset accounts and the hidden account isn't necessary. That's the difference.

A liability account is not really the loan itself, it's the account that holds the amount that you *then* loan to a friend (a friend being an expense account, basically). You have to keep track of the loan on a separate account to make it easier to have multiple destinations for the money and multiple sources for people paying you back.

You could do transfers instead of deposits/withdrawals between liabilities. After all, the loan only changes when you move the money to a friend. But that doesn't reflect the fact that if you pay back a loan the money is "spent".

**Reply by OXINARF** [reply]
I had not thought about net worth, that's an excellent point. I've updated the table above with that and with how the initial debt works.

Your explanation makes sense, although you could apply to it to when you owe (mortgage for example). If I start Firefly III with a mortgage of 1000, my net worth starts at -1000, meaning I'm owing 1000. Once the mortgage is paid back, I'm back at zero - but that's only if I only record "Asset to Liability" transaction and not a "Liability to Expense" transaction too (which would be similar to what you do with when it is that are owed).
I definitely understand why you want to start with negative net worth when you have a mortgage, you don't have that money, you'll have to pay it back. But the same applies when it is you giving a loan - that money is yours and the person that you loaned to will have to pay you back, so you could have that amount added to net worth (and not subtracted); then you would do as you do when you owe and only record "Liability to Asset" transactions.

In the end, it is just strange (to me) that things are handled differently whether you are giving a loan or receiving one - in my perspective, they should basically be the opposite of each other, but they aren't.

**Reply by JC5** [reply]
> Once the mortgage is paid back, I'm back at zero - but that's only if I only record "Asset to Liability" transaction and not a "Liability to Expense" transaction too (which would be similar to what you do with when it is that are owed).

I just tried this, but this is false. Unless your net worth is increased to zero by a deposit of 1000 from an external revenue account, your net worth remains -1000 if you pay back the mortgage.

> But the same applies when it is you giving a loan - that money is yours and the person that you loaned to will have to pay you back, so you could have that amount added to net worth (and not subtracted); then you would do as you do when you owe and only record "Liability to Asset" transactions.

Yes, this is typically what accountants do. But since you don't have that money, Firefly III doesn't add it to your net worth.

**Comment by tmllull** [comment]
Hi! I'm using this post because of my question is related to the title.

I'm looking to use a liabilities option, and I'm doing some tests on my "testing" instance before, and I've 2 questions:
1. I understand the "money flow" between the accounts (asset and liabilities accounts). I start a new liability account (John's travel) with "-€400", because a friend owe some money to me (I paid some stuff in a travel). Few days later this friend pays part of this debt (€100), so I add a new deposit into "John's travel" account, and then I transfer this money to my main account. Why is this transfer green instead of blue?
![imagen](https://user-images.githubusercontent.com/5560652/204761962-a39fb259-f1b9-49f0-b868-640b60c1c7d9.png)

2. My other question is about the "total amount" for the resume of this account. If I go to the transactions details for John's travel account, shows that I earned €200, but actually I only earned 100 (the other 100 are from the transfer). Is this correct?
![imagen](https://user-images.githubusercontent.com/5560652/204762698-078eb4fe-ce8c-426a-b243-2af1d676eff5.png)

Thanks for your help! You're doing an amazing app!

**Reply by JC5** [reply]
> Why is this transfer green instead of blue?

Because it becomes a deposit into your asset account when it moves from the liability to you. The money is not yet ready to be used by you until you put it in your asset account.

> My other question is about the "total amount" for the resume of this account.

Yeah that's not correct, and something I hope to fix in the new layout.

**Comment by OXINARF** [comment]
@JC5 I saw the alpha release for 5.8 with changes to Liabilities, so there I went testing it on demo website. This how the table is now:
<table>
<thead>
  <tr>
    <th></th>
    <th align="center" valign="middle" colspan="3">Owe</th>
    <th align="center" valign="middle" colspan="3">Owed</th>
  </tr>
</thead>
<tbody>
  <tr>
    <th></th>
    <th align="center" valign="middle">Balance</th>
    <th align="center" valign="middle">Amount due</th>
    <th align="center" valign="middle">Net worth</th>
    <th align="center" valign="middle">Balance</th>
    <th align="center" valign="middle">Amount due</th>
    <th align="center" valign="middle">Net worth</th>
  </tr>
  <tr>
    <td valign="middle">Liability to Expense (withdrawal)</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Goes positive</td>
    <td align="center" valign="middle">Doesn't change</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Goes positive</td>
    <td align="center" valign="middle">Doesn't change</td>
  </tr>
  <tr>
    <td valign="middle">Revenue to Liability (deposit)</td>
    <td align="center" valign="middle">Goes positive</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Doesn't change</td>
    <td align="center" valign="middle">Goes positive</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Doesn't change</td>
  </tr>
  <tr>
    <td valign="middle">Asset to Liability (withdrawal)</td>
    <td align="center" valign="middle">Goes positive</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Goes positive</td>
    <td align="center" valign="middle">Goes positive</td>
    <td align="center" valign="middle">Goes negative</td>
  </tr>
  <tr>
    <td valign="middle">Liability to Asset (deposit)</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Goes positive</td>
    <td align="center" valign="middle">Goes positive</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Goes negative</td>
    <td align="center" valign="middle">Goes positive</td>
  </tr>
  <tr>
    <td colspan="7"></td>
  </tr>
  <tr>
    <td valign="middle" rowspan="2">Initial debt</td>
    <th align="center" valign="middle" colspan="3">Liability to Liability Credit transaction</th>
    <th align="center" valign="middle" colspan="3">Liability Credit to Liability transaction</th>
  </tr>
  <tr>
    <td align="center" valign="middle">Negative</td>
    <td align="center" valign="middle">Positive</td>
    <td align="center" valign="middle">Doesn't change</td>
    <td align="center" valign="middle">Positive</td>
    <td align="center" valign="middle">Positive</td>
    <td align="center" valign="middle">Doesn't change</td>
  </tr>
</tbody>
</table>

I don't understand the Owed type now - it seems to me that using transactions between Liability and Expense/Renevue accounts would never make sense: should it be forbidden or is it a bug? I think that "Amount due" shouldn't change on those.

By the way, there is still the "Include in Net worth" button on Liability create/edit page, which doesn't do anything now.

What will happen with existing liabilities when we upgrade? I currently have long-term Owed liabilities where I have the dual transaction scheme (so that balance is 0) and a short-term Owed liability where I only make transactions between Asset accounts and Liability (this makes balance occasionally bigger than 0, but that's exactly what I intended, that money keeps being part of my Net Worth, it will be returned very soon).

With this update, it seems my both my use-cases will no longer be possible... Can you explain what was the thought behind these changes? Not including in Net Worth was already achievable by disabling the button, right? As for the dual transaction for Owed type, it seems to me that your changes are good, except the issue with the Liability<->Expense/Revenue transactions (which, if fixed, should have no impact on using the single transaction scheme).
